pg_search/src/fixtures/index.rs

```diff
@@ -28,11 +31,21 @@ pub struct MockSearchIndex {
 }
 
 impl MockSearchIndex {
-    pub fn new(fields: Vec<(SearchFieldName, SearchFieldConfig, SearchFieldType)>) -> Self {
+    pub fn new(
+        fields: Vec<(SearchFieldName, SearchFieldConfig, SearchFieldType)>,
+        key_field_index: usize,
+    ) -> Self {
         // We must store the TempDir instance on the struct, because it gets deleted when the
         // instance is dropped.
         let directory = MockWriterDirectory::new("mock_parade_search_index");
-        let index = SearchIndex::new(directory.writer_dir.clone(), fields).unwrap();
+        let mut writer = Writer::new();
+        let uuid = Uuid::new_v4().to_string();
+        writer
+            .create_index(directory.writer_dir.clone(), fields, uuid, key_field_index)
```

Comment:
Question: what is the purpose of `key_field_index`?

Reply:
In the adoption of non-integer `key_field` values, @sardination introduced a `SearchFieldConfig:: Key(Box<SearchFieldConfig>)` variant. This labelled one of the `SearchFieldConfig` values as the `key_field`, so the background writer knows what the key is.

That kind of recursive enum definition is not supported by Bincode. Now that we have to send the `SearchFieldConfig` values across the IPC boundary to the background writer, we need the `SearchFieldConfig` to be Bincode-compatible.

So I've removed the `SearchFieldConfig::Key` variant to simplify the enum structure. We just need another way to tell the background writer which `Vec<SearchFieldConfig>` values it receives is the `key_field`. The `key_field_index` is that piece of information.